packages/blade/src/components/Input/BaseInput/BaseInputVisuals.tsx

```diff
@@ -314,24 +317,30 @@ export const BaseInputVisuals = ({
     );
   }
 
+  if (hasTrailingVisuals && hasTrailingInteractionElement) {
```

Comment:
This block ends up early returning the jsx unlike the previous implementation. So when there is trailingInteractionElement + trailing visual, it seems to only render trailingInteractionElement and skip the code that comes after this.

You can see this in this example where when showClearButton is set to true, and you start typing, it removes the trailing visual button https://61c19ee8d3d282003ac1d81c-kxotdzpypj.chromatic.com/?path=/story/components-input-textinput--text-input-with-trailing-button&args=showClearButton:!true&globals=showInternalComponents:!true


First is behaviour in this PR, and second is current production behaviour

https://github.com/user-attachments/assets/b3d8c57c-2423-4d8c-96ed-b7e50e30b106